src/UsgsAstroLsSensorModel.cpp

```diff
@@ -2664,57 +2660,56 @@ std::string UsgsAstroLsSensorModel::constructStateFromIsd(const std::string imag
-     state["m_intTimes"] = {isd.at("INT_TIME").get<double>()};
+     state["m_intTimes"] = {isd.at("int_time").get<double>()};
+     state["m_intTimeStartTimes"] = {state["m_totalLines"].get<double>() * state["m_intTimes"][0].get<double>() / 2 };
    }
    else {
-     int numIntTimes = isd.at("NUMBER_OF_INT_TIMES");
+     int numIntTimes = isd.at("number_of_int_times");
      for (int i = 0; i < numIntTimes; i++) {
-       state["m_intTimeLines"].push_back(isd.at("INT_TIME").at(i*3));
-       state["m_intTimeStartTimes"].push_back(isd.at("INT_TIME").at(i*3+1));
-       state["m_intTimes"].push_back(isd.at("INT_TIME").at(i*3+2));
+       // TODO what does this look like under the new spec?
+       state["m_intTimeLines"].push_back(isd.at("int_time").at(i*3));
+       state["m_intTimeStartTimes"].push_back(isd.at("int_time").at(i*3+1));
+       state["m_intTimes"].push_back(isd.at("int_time").at(i*3+2));
      }
    }
 
-   state["m_centerEphemerisTime"] = isd.at("CENTER_EPHEMERIS_TIME");
-   state["m_detectorSampleSumming"] = isd.at("DETECTOR_SAMPLE_SUMMING");
-   state["m_startingSample"] = isd.at("STARTING_SAMPLE");
-   state["m_ikCode"] = isd.at("IKCODE");
-   state["m_focal"] = isd.at("FOCAL");
-   state["m_isisZDirection"] = isd.at("ISIS_Z_DIRECTION");
+   state["m_startingEphemerisTime"] = state["m_intTimes"].at(0);
+   state["m_detectorSampleSumming"] = 1;
+   state["m_startingSample"] = 0;
+   state["m_ikCode"] = 0;
+   state["m_focal"] = isd.at("focal_length_model").at("focal_length");
+   state["m_isisZDirection"] = 1;
 
    for (int i = 0; i < 3; i++)
    {
-      state["m_opticalDistCoef"][i] = isd.at("OPTICAL_DIST_COEF").at(i);
-      state["m_iTransS"][i] = isd.at("ITRANSS").at(i);
-      state["m_iTransL"][i] = isd.at("ITRANSL").at(i);
+      state["m_opticalDistCoef"][i] = isd.at("optical_distortion").at("r").at(i);
+      state["m_iTransS"][i] = isd.at("focal2pixel_samples").at(i);
+      state["m_iTransL"][i] = isd.at("focal2pixel_lines").at(i);
    }
 
-   state["m_detectorSampleOrigin"] = isd.at("DETECTOR_SAMPLE_ORIGIN");
-   state["m_detectorLineOrigin"] = isd.at("DETECTOR_LINE_ORIGIN");
-   state["m_detectorLineOffset"] = isd.at("DETECTOR_LINE_OFFSET");
+   state["m_detectorSampleOrigin"] = isd.at("detector_center").at("sample");
+   state["m_detectorLineOrigin"] = isd.at("detector_center").at("line");
+   state["m_detectorLineOffset"] = 0;
 
-   double cos_a = cos(isd.at("MOUNTING_ANGLES").at(0).get<double>());
-   double sin_a = sin(isd.at("MOUNTING_ANGLES").at(0).get<double>());
-   double cos_b = cos(isd.at("MOUNTING_ANGLES").at(1).get<double>());
-   double sin_b = sin(isd.at("MOUNTING_ANGLES").at(1).get<double>());
-   double cos_c = cos(isd.at("MOUNTING_ANGLES").at(2).get<double>());
-   double sin_c = sin(isd.at("MOUNTING_ANGLES").at(2).get<double>());
+   double cos_a = cos(0);
+   double sin_a = sin(0);
+   double cos_b = cos(0);
+   double sin_b = sin(0);
+   double cos_c = cos(0);
+   double sin_c = sin(0);

```

Comment:
:+1:

Reply:
This part will get removed eventually so for right now just sets all the mounting angles to 0